lib/codegen/generator-soap.js

```diff
@@ -360,9 +360,7 @@ function buildElementProperties(models, modelName,
         // could be a complextype or xs: simple type
         type = element.type.name;
       }
-      propertyData = {
-        type: checkAndConvertToNumber(type),
-      };
+      propertyData = createPropertyData(checkAndConvertToNumber(type), element);
```

Comment:
@emonddr is this change related to the issue, or are we addressing some additional concerns ?

Reply:
@deepakrkris , 

```
 propertyData = {
        type: checkAndConvertToNumber(type),
      };
```

was creating an object, 

```
{ type: 'string' ( or 'number' or 'MyModel' ) }
```

I needed a way to create propertyData that could ALSO create:

```
{ type: ['string'] ( or ['number'] or '[MyModel'] ) }

```

So instead of creating an `if else block` at this point in the code ; above the following line

```
 propertyList[element.qname.name] = propertyData;
```

I chose to create a new function `createPropertyData()` which takes in the
calculated type from checkAndConvertToNumber(), and can produce either of the scenarios above, based on the isMany value of the original xsd element loaded from the wsdl file

```
function createPropertyData(typeName, element) {
  return (element.isMany && element.isMany == true) ?
    {type: [typeName]} : {type: typeName};
}
```

Reply:
@deepakrkris , I responded to your comments. ^